src/kaocha/repl.clj

```diff
@@ -151,11 +153,17 @@ These will particularly come in handy when developing plugins."}
                                [{} args])
          config (-> (config config-opts)
                     (update :kaocha.filter/focus into (map testable-id) tests))]
-     (->> config
-          api/run
-          (plugin/run-hook :kaocha.hooks/post-summary)
-          :kaocha.result/tests
-          result/totals))))
+     (try+
+      (->> config
+           api/run
+           (plugin/run-hook :kaocha.hooks/post-summary)
+           :kaocha.result/tests
+           result/totals)
+      (catch :kaocha/early-exit {exit-code :kaocha/early-exit}
+        (if (not= exit-code 0)
+          (output/error "Test run exited with code " exit-code)
+          (output/warn "Test run exited with code " exit-code))
```

Comment:
do we ever `:early-exit` with code 0? Seems contradictory.

Reply:
Yeah, probably. We do if no tests are found (its own issue #211) and in the `kaocha.plugin.alpha.info` plugin.